Subject: powerpc/ps3: Update PS3 linker script

Update the PS3 linker script to follow the changes in the
powerpc wrapper to build the boot image as PIC.

Signed-off-by: Geoff Levand <geoffrey.levand@am.sony.com>
---
 arch/powerpc/boot/zImage.ps3.lds.S |   12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

--- a/arch/powerpc/boot/zImage.ps3.lds.S
+++ b/arch/powerpc/boot/zImage.ps3.lds.S
@@ -8,15 +8,11 @@ SECTIONS
   _vmlinux_end =  .;
 
   . = ALIGN(4096);
-  _dtb_start = .;
-  .kernel:dtb : { *(.kernel:dtb) }
-  _dtb_end = .;
-
-  . = ALIGN(4096);
   _initrd_start =  .;
   .kernel:initrd : { *(.kernel:initrd) }
   _initrd_end =  .;
 
+  . = ALIGN(4096);
   _start = .;
   .text      :
   {
@@ -24,6 +20,7 @@ SECTIONS
     *(.fixup)
   }
   _etext = .;
+
   . = ALIGN(4096);
   .data    :
   {
@@ -35,6 +32,11 @@ SECTIONS
     __got2_end = .;
   }
 
+  . = ALIGN(8);
+  _dtb_start = .;
+  .kernel:dtb : { *(.kernel:dtb) }
+  _dtb_end = .;
+
   . = ALIGN(4096);
   _edata  =  .;
 
